gallery/__init__.py

```diff
@@ -876,21 +959,22 @@ def get_dir_children(dir_id: int) -> Any:
 @app.route("/api/directory/get/<int:dir_id>")
 @auth.oidc_auth('default')
 @gallery_auth
-def display_files(dir_id: int, internal: bool = False, auth_dict: Optional[Dict[str, Any]] = None):
+def display_files(dir_id: int, internal: bool = False, auth_dict: Optional[Dict[str, Any]] = None, pinned = False):
     gallery_lockdown = util.get_lockdown_status()
     if gallery_lockdown and (not auth_dict['is_eboard'] and not auth_dict['is_rtp']):
         abort(405)
 
-    file_list = [("File", f) for f in File.query.filter(File.parent == dir_id).all()]
-    dir_list = [("Directory", d) for d in Directory.query.filter(Directory.parent == dir_id).all()]
+    file_list = [("File", f) for f in File.query.filter(File.parent == dir_id, File.pinned == pinned).all()]
+    dir_list = [("Directory", d) for d in Directory.query.filter(Directory.parent == dir_id, Directory.pinned == pinned).all()]
 
     # Sort by name/title
     file_list.sort(key=lambda x: x[1].get_name())
     file_list.sort(key=lambda x: x[1].date_uploaded)
     dir_list.sort(key=lambda x: x[1].get_name())
     dir_list.sort(key=lambda x: x[1].date_uploaded)
 
-    ret_dict = dir_list + file_list
+    ret_dict = file_list + dir_list
```

Comment:
Why swap these?

Reply:
Whoops, I accidentally deleted line and thought I remembered the order.